Example/Custom Integration/CardSetupIntentExampleViewController.m

```diff
@@ -0,0 +1,139 @@
+ 
+ 1. Collect user's card information via `STPPaymentCardTextField`
+ 2. Create a `SetupIntent` on our backend (this can happen concurrently with #1)
+ 3. Confirm SetupIntent with `STPPaymentHandler`, using the `STPPaymentMethodParams` for the user's card information.
+ */
+@interface CardSetupIntentExampleViewController () <STPPaymentCardTextFieldDelegate>
+
+@property (weak, nonatomic) STPPaymentCardTextField *paymentTextField;
+@property (weak, nonatomic) UILabel *waitingLabel;
+@property (weak, nonatomic) UIActivityIndicatorView *activityIndicator;
+
+@end
+
+@implementation CardSetupIntentExampleViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.view.backgroundColor = [UIColor whiteColor];
+    self.title = @"Card";
+    self.edgesForExtendedLayout = UIRectEdgeNone;
+    
+    STPPaymentCardTextField *paymentTextField = [[STPPaymentCardTextField alloc] init];
+    STPPaymentMethodCardParams *cardParams = [STPPaymentMethodCardParams new];
+    // Only successful 3D Secure transactions on this test card will succeed.
+    cardParams.number = @"4000000000003063";
+    paymentTextField.cardParams = cardParams;
+    paymentTextField.delegate = self;
+    paymentTextField.cursorColor = [UIColor purpleColor];
+    self.paymentTextField = paymentTextField;
+    [self.view addSubview:paymentTextField];
+    
+    UILabel *label = [UILabel new];
+    label.text = @"Waiting for payment authorization";
+    [label sizeToFit];
+    label.textColor = [UIColor grayColor];
+    label.alpha = 0;
+    [self.view addSubview:label];
+    self.waitingLabel = label;
+    
+    NSString *title = @"Pay";
+    UIBarButtonItem *payButton = [[UIBarButtonItem alloc] initWithTitle:title style:UIBarButtonItemStyleDone target:self action:@selector(pay)];
+    payButton.enabled = paymentTextField.isValid;
+    self.navigationItem.rightBarButtonItem = payButton;
+    
+    UIActivityIndicatorView *activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
+    activityIndicator.hidesWhenStopped = YES;
+    self.activityIndicator = activityIndicator;
+    [self.view addSubview:activityIndicator];
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    [self.paymentTextField becomeFirstResponder];
+}
+
+- (void)viewDidLayoutSubviews {
+    [super viewDidLayoutSubviews];
+    CGFloat padding = 15;
+    CGFloat width = CGRectGetWidth(self.view.frame) - (padding*2);
+    CGRect bounds = self.view.bounds;
+    self.paymentTextField.frame = CGRectMake(padding, padding, width, 44);
+    self.activityIndicator.center = CGPointMake(CGRectGetMidX(bounds),
+                                                CGRectGetMaxY(self.paymentTextField.frame) + padding*2);
+    self.waitingLabel.center = CGPointMake(CGRectGetMidX(bounds),
+                                           CGRectGetMaxY(self.activityIndicator.frame) + padding*2);
```

Comment:
manual layout haha

Reply:
heh, copy and pasted from one of the other VCs.  I'd like to make some other improvements to this app at some point..